README.md

```diff
@@ -107,49 +83,98 @@ Add required providers using the utility function `subformComponentProviders`:
+-export class VehicleListingComponent {}
++export class VehicleListingComponent extends NgxSubFormComponent {}
 ```
 
 Define the controls of your form:
 
 ```ts
-private controls: Controls<VehiculeSell> = {
-  id: new FormControl(this.uuidService.generate(), { validators: [Validators.required] }),
-  price: new FormControl(null, { validators: [Validators.required] }),
+protected formControls: Controls<VehicleListing> = {
+  id: new FormControl(this.uuidService.generate(), Validators.required),
+  price: new FormControl(null, Validators.required),
 };
 
-public formGroup: FormGroup = new FormGroup(this.controls);
-
-public controlsNames: ControlsNames<VehiculeSell> = getControlsNames(this.controls);
 ```
 
 _Simplified from the original example into src folder to keep the example as minimal as possible._
 
 As you know, [Angular reactive forms are not strongly typed](https://github.com/angular/angular/issues/13721). We're providing an interface (`Controls<T>`) to at least set the correct names within the form (but it will not help you when using `form.get('...').value`). It is still very useful and when making a refactor if your data structure changes and do not match the form structure Typescript compilation will fail.
 
-We also provide a utility function called `getControlsNames` which you can pass your `controls` to. This will let you define your `formControlName`s in the view in a safe way thanks to AoT. If you update your main interface, your form control but you forget about the view, you'll get an error.
+The NgxFormComponent base class automatically extracts your form control names and exposes them as a public member on `formControlNames`
 
-Then within the `.html`:
+Then within the `.html`, you can reference them like so:
 
 ```html
 <fieldset [formGroup]="formGroup" class="container">
-  <input type="text" placeholder="ID" [formControlName]="controlsNames.id" />
+  <input type="text" placeholder="ID" [formControlName]="formControlNames.id" />
 
-  <input type="number" placeholder="Price" [formControlName]="controlsNames.price" />
+  <input type="number" placeholder="Price" [formControlName]="formControlNames.price" />
 </fieldset>
 ```
+
+### Remapping Data
+
+It is a frequent pattern to have the data that you're trying to modify in a format that is incovenient to the angular
+forms structural constraints. For this reason, ngx-form-component offers a separate extended class `NgxSubFormRemapComponent`
+which will require you to define two interfaces - one to model the data going in to the form (which will be applied
+internally as `form.setValue()`), and the other to describe the interface of the value that will be set on the form.
+
+**You're always better off making your data structure better suit Angular forms, than abusing forms to fit your data pattern**
+
+For a complete example of this see `src/app/main/listing/vehicle-listing/vehicle-product.component.ts` (repeated below):
```

Comment:
set is as a link with https://github.com/cloudnc/ngx-sub-form/blob/master/src/app/main/listing/vehicle-listing/vehicle-product.component.ts